File: Segunda Practica Auxiliatura /Herencia/Eje1/eje1.py
# INCISO A

class Vehiculo:
    def __init__(self, marca, modelo, anio, precio_base):
        self._marca = marca
        self._modelo = modelo
        self._anio = anio
        self._precio_base = precio_base

    def get_marca(self):
        return self._marca

    def get_modelo(self):
        return self._modelo

    def get_anio(self):
        return self._anio

    def get_precio_base(self):
        return self._precio_base

    def set_marca(self, marca):
        self._marca = marca

    def set_modelo(self, modelo):
        self._modelo = modelo

    def set_anio(self, anio):
        self._anio = anio

    def set_precio_base(self, precio):
        self._precio_base = precio

    def mostrar_info(self):
        print(f"Marca: {self._marca}")
        print(f"Modelo: {self._modelo}")
        print(f"Año: {self._anio}")
        print(f"Precio base: ${self._precio_base}")


class Coche(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, num_puertas, tipo_combustible):
        super().__init__(marca, modelo, anio, precio_base)
        self._num_puertas = num_puertas
        self._tipo_combustible = tipo_combustible

    def get_num_puertas(self):
        return self._num_puertas

    def get_tipo_combustible(self):
        return self._tipo_combustible

    def set_num_puertas(self, num_puertas):
        self._num_puertas = num_puertas

    def set_tipo_combustible(self, tipo_combustible):
        self._tipo_combustible = tipo_combustible

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Número de puertas: {self._num_puertas}")
        print(f"Tipo de combustible: {self._tipo_combustible}")


class Moto(Vehiculo):
    def __init__(self, marca, modelo, anio, precio_base, cilindrada, tipo_motor):
        super().__init__(marca, modelo, anio, precio_base)
        self._cilindrada = cilindrada
        self._tipo_motor = tipo_motor

    def get_cilindrada(self):
        return self._cilindrada

    def get_tipo_motor(self):
        return self._tipo_motor

    def set_cilindrada(self, cilindrada):
        self._cilindrada = cilindrada

    def set_tipo_motor(self, tipo_motor):
        self._tipo_motor = tipo_motor

    def mostrar_info(self):
        super().mostrar_info()
        print(f"Cilindrada: {self._cilindrada}cc")
        print(f"Tipo de motor: {self._tipo_motor}")


# INCISO B

vehiculos = [
    Coche("Toyota", "Corolla", 2025, 22000, 4, "Gasolina"),
    Coche("Chevrolet", "Spin", 2023, 25000, 5, "Gasolina"),
    Coche("BMW", "Serie 3", 2025, 35000, 2, "Diésel"),
    Moto("Yamaha", "R3", 2025, 12000, 321, "4 tiempos"),
    Moto("Honda", "CBR500R", 2024, 15000, 471, "4 tiempos")
]

print("Información de todos los vehículos:\n")
for v in vehiculos:
    v.mostrar_info()
    print("------------")


# INCISO C

print("Coches con más de 4 puertas:\n")
for v in vehiculos:
    if isinstance(v, Coche) and v.get_num_puertas() > 4:
        v.mostrar_info()
        print("------------")


# INCISO D

print("Vehículos del año 2025:\n")
for v in vehiculos:
    if v.get_anio() == 2025:
        v.mostrar_info()
        print("------------")

File: Segunda Practica Auxiliatura /Herencia/Eje1/Eje1.java
// INCISO A

class Vehiculo {
    private String marca;
    private String modelo;
    private int anio;
    private double precioBase;

    public Vehiculo(String marca, String modelo, int anio, double precioBase) {
        this.marca = marca;
        this.modelo = modelo;
        this.anio = anio;
        this.precioBase = precioBase;
    }

    public String getMarca() {
        return marca;
    }

    public String getModelo() {
        return modelo;
    }

    public int getAnio() {
        return anio;
    }

    public double getPrecioBase() {
        return precioBase;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public void setAnio(int anio) {
        this.anio = anio;
    }

    public void setPrecioBase(double precioBase) {
        this.precioBase = precioBase;
    }

    public void mostrarInfo() {
        System.out.println("Marca: " + marca);
        System.out.println("Modelo: " + modelo);
        System.out.println("Año: " + anio);
        System.out.println("Precio base: $" + precioBase);
    }
}

class Coche extends Vehiculo {
    private int numPuertas;
    private String tipoCombustible;

    public Coche(String marca, String modelo, int anio, double precioBase, int numPuertas, String tipoCombustible) {
        super(marca, modelo, anio, precioBase);
        this.numPuertas = numPuertas;
        this.tipoCombustible = tipoCombustible;
    }

    public int getNumPuertas() {
        return numPuertas;
    }

    public String getTipoCombustible() {
        return tipoCombustible;
    }

    public void setNumPuertas(int numPuertas) {
        this.numPuertas = numPuertas;
    }

    public void setTipoCombustible(String tipoCombustible) {
        this.tipoCombustible = tipoCombustible;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Número de puertas: " + numPuertas);
        System.out.println("Tipo de combustible: " + tipoCombustible);
    }
}

class Moto extends Vehiculo {
    private int cilindrada;
    private String tipoMotor;

    public Moto(String marca, String modelo, int anio, double precioBase, int cilindrada, String tipoMotor) {
        super(marca, modelo, anio, precioBase);
        this.cilindrada = cilindrada;
        this.tipoMotor = tipoMotor;
    }

    public int getCilindrada() {
        return cilindrada;
    }

    public String getTipoMotor() {
        return tipoMotor;
    }

    public void setCilindrada(int cilindrada) {
        this.cilindrada = cilindrada;
    }

    public void setTipoMotor(String tipoMotor) {
        this.tipoMotor = tipoMotor;
    }

    @Override
    public void mostrarInfo() {
        super.mostrarInfo();
        System.out.println("Cilindrada: " + cilindrada + "cc");
        System.out.println("Tipo de motor: " + tipoMotor);
    }
}

public class Main {
    public static void main(String[] args) {
        Vehiculo[] vehiculos = {
            new Coche("Toyota", "Corolla", 2025, 22000, 4, "Gasolina"),
            new Coche("Chevrolet", "Spin", 2023, 25000, 5, "Gasolina"),
            new Coche("BMW", "Serie 3", 2025, 35000, 2, "Diésel"),
            new Moto("Yamaha", "R3", 2025, 12000, 321, "4 tiempos"),
            new Moto("Honda", "CBR500R", 2024, 15000, 471, "4 tiempos")
        };

        // INCISO B
        System.out.println("Información de todos los vehículos:\n");
        for (Vehiculo v : vehiculos) {
            v.mostrarInfo();
            System.out.println("------------");
        }

        // INCISO C
        System.out.println("Coches con más de 4 puertas:\n");
        for (Vehiculo v : vehiculos) {
            if (v instanceof Coche && ((Coche) v).getNumPuertas() > 4) {
                v.mostrarInfo();
                System.out.println("------------");
            }
        }

        // INCISO D
        System.out.println("Vehículos del año 2025:\n");
        for (Vehiculo v : vehiculos) {
            if (v.getAnio() == 2025) {
                v.mostrarInfo();
                System.out.println("------------");
            }
        }
    }
}
